backend/views.py

```diff
@@ -1,8 +1,36 @@
-from canvas_oauth.oauth import get_oauth_token
+import logging
+
+from canvas_oauth.oauth import get_oauth_token, handle_missing_token
+from canvas_oauth.models import CanvasOAuth2Token
+from canvas_oauth.exceptions import CanvasOAuthError, MissingTokenError, InvalidOAuthStateError, InvalidOAuthReturnError
+
+from django.contrib.auth.models import User
 from django.contrib.auth.decorators import login_required
 from django.shortcuts import render
 
+
+logger = logging.getLogger(__name__)
+
 @login_required
 def get_home_template(request):
-    access_token = get_oauth_token(request)
+    try:
+        access_token = get_oauth_token(request)
+    except InvalidOAuthStateError as err:
+        logger.error("InvalidOAuthStateError. Remove invalid refresh_token and prompt for reauthentication.")
+        CanvasOAuth2Token.objects.filter(user=request.user).delete()
+        return handle_missing_token(request)
```

Comment:
This one may also be unnecessary, since this is part of the original OAuth handshake and relates to the state parameter and prevention of CSRF. I don't think we'd want to delete the token in this case.

Reply:
removed this error handling.